docs/source/general/how_to_extend_django_freeradius.rst

```diff
@@ -0,0 +1,114 @@
+===============================
+How to extend django-freeradius
+===============================
+
+``django-freeeadius`` provieds set of models, admin and API classes which can be imported, extended and hence customized by third party apps.
+
+Update Settings
+---------------
+
+Update the settings to trigger the swapper:
+
+.. code-block:: python
+
+    # In settings.py of your project
+        
+    DJANGO_FREERADIUS_RADIUSREPLY_MODEL = "my_radius_app.RadiusReply"
+    DJANGO_FREERADIUS_RADIUSGROUPREPLY_MODEL = "my_radius_app.RadiusGroupReply"
+    DJANGO_FREERADIUS_RADIUSCHECK_MODEL = "my_radius_app.RadiusCheck"
+    DJANGO_FREERADIUS_RADIUSGROUPCHECK_MODEL = "my_radius_app.RadiusGroupCheck"
+    DJANGO_FREERADIUS_RADIUSACCOUNTING_MODEL = "my_radius_app.RadiusAccounting"
+    DJANGO_FREERADIUS_NAS_MODEL = "my_radius_app.Nas"
+    DJANGO_FREERADIUS_RADIUSUSERGROUP_MODEL = "my_radius_app.RadiusUserGroup"
+    DJANGO_FREERADIUS_RADIUSPOSTAUTHENTICATION_MODEL = "my_radius_app.RadiusPostAuth"
+
+    # where my_radius_app is name of your app extending django_freeradius
+
+
+Extend models
+-------------
+
+Apart from extending implemented models, ``django_freeradius`` also provides flexibility to extend abstract class models from `django-freeradius.base.models`.
+
+Example:
+
+.. code-block:: python
+
+    # In my_radius_app/models.py
+
+    from django.db import models
+    from django_freeradius.base.models import AbstractRadiusCheck
+    
+    class RadiusCheck(AbstractRadiusCheck):
+        # modify/extend the default behavour here
+        custom_field = models.TextField()
+
```

Comment:
since in the settings part you have put all the models, you should put all the models here too, you may copy them from openwisp-radius.

Reply:
That might make documentation a little lengthy, what about leaving a note to do similarly for others?

Reply:
@TheOneAboveAllTitan that's ok too but keep in mind code samples are usually copied verbatim and skipped by readers, so if they're long because they need to be long is not a big deal. Code is necessary some times. What we have to put effort in keeping short is the text, being concise and straight to the point with as less words as possible, that's more important than omitting part of code examples that are required to achieve the desired goal effectively (keep in mind some user smay not know django very well or not at all and may not understand the docs if examples are not provided in full).